src/io.rs

```diff
@@ -805,22 +826,27 @@ impl FdOutputStream {
     }
 }
 
-/// A simple output stream which buffers into a wcstring.
-#[derive(Default)]
-pub struct StringOutputStream {
-    contents: WString,
-}
-impl StringOutputStream {
-    pub fn new() -> Self {
-        Default::default()
-    }
-    fn append(&mut self, s: &wstr) -> bool {
-        self.contents.push_utfstr(s);
-        true
+impl io::Write for FdOutputStream {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
```

Comment:
note that this doesn't match FdOutputStream::append() which does nothing if the stream errored previously.
Maybe we should move the error check to the callers so we can implement the std::io::Write
interface without surprising changes. But that may be too noisy.